bigquery/google/cloud/bigquery/table.py

```diff
@@ -741,11 +741,9 @@ def insert_data(self,
             row_info = {}
 
             for field, value in zip(self._schema, row):
-                if field.field_type == 'TIMESTAMP':
-                    # BigQuery stores TIMESTAMP data internally as a
-                    # UNIX timestamp with microsecond precision.
-                    # Specifies the number of seconds since the epoch.
-                    value = _convert_timestamp(value)
+                converter = _SCALAR_VALUE_TO_JSON_ROW.get(field.field_type)
+                if converter is not None:
```

Comment:
Is your concern about performance?  If so, the branch test is going to be orders of magnitude cheaper than a pointless function call.